modules/frontend/query_range_sharding.go

```diff
@@ -387,8 +395,7 @@ func (s *queryRangeSharder) backendRange(now time.Time, start, end uint64, query
 	return start, end
 }
 
-func (s *queryRangeSharder) generatorRequest(searchReq tempopb.QueryRangeRequest, samplingRate float64) *queryRangeJob {
-	now := time.Now()
+func (s *queryRangeSharder) generatorRequest(searchReq tempopb.QueryRangeRequest, now time.Time, samplingRate float64) *queryRangeJob {
```

Comment:
`now` seems to be unused

Reply:
It's used [a bit further down](https://github.com/grafana/tempo/pull/3490/files/0432ccee3ed4f790c4b71415463194ff4bb39bfa#diff-1205a2cd25b76a20cc27dc681f5073614b170b1dfbf29069af221e6f8e7ffed6R399) to determine the starting time range of the generator request: now-querybackendafter.